pandas/io/excel/_base.py

```diff
@@ -1503,7 +1509,13 @@ def __init__(
 
         # First argument can also be bytes, so create a buffer
         if isinstance(path_or_buffer, bytes):
-            path_or_buffer = BytesIO(path_or_buffer)
```

Comment:
To maintain the current behavior during the deprecation, I think we need to continue wrapping this in `BytesIO`

Reply:
@mroeschke What is the action here? I'm wrapping it in BytesIO

Reply:
Ah yep! I added this back. Thanks for catching this!

Reply:
It seems that `test_excel_read_binary` still passes without this line on main. If we are wanting to remove this functionality so that we raise when bytes are passed, I think more needs to be done. But I think this can wait until the deprecation is enforced.

Reply:
@rhshadrach I'm probably missing something, but that test is passing because the input to `read_excel` for that particular test is wrapped in a BytesIO object before it hits that if statement. 

If I were to remove the line of code that casts to a BytesIO object in both the test and in _base.py, the test will fail.

Reply:
@rhshadrach I'm a bit confused on your above message :thinking: 

The test is passing without the line on main because the input to `read_excel` is already a BytesIO object. If I left the input as raw binary the test would fail.

Reply:
The test in question is

```
        expected = pd.read_excel("test1" + read_ext, engine=engine)

        with open("test1" + read_ext, "rb") as f:
            data = f.read()

        actual = pd.read_excel(data, engine=engine)
        tm.assert_frame_equal(expected, actual)
```

`data` here is a bytes, not a BytesIO, object.

Reply:
@rhshadrach Matthew has been helping me with a similar problem on a few other PRs.

Are you okay with it if I go ahead and take care of this discrepancy before we submit this one?

Reply:
I'm pretty sure the fix for this problem is just adding more input validation by using a handful of methods from common.py

Reply:
Agreed, I think we're okay to deprecate with this PR as is, but this can wait until the deprecation is enforced (i.e. the feature of reading bytes is removed in pandas 3.0).